lib/google_sheets/config.rb

```diff
@@ -0,0 +1,37 @@
+# Author: Mateusz Czerwinski <mtczerwinski@gmail.com>
+# The license of this source is "New BSD Licence"
+# Copied from https://github.com/gimite/google-drive-ruby
+
+require 'json'
+
+module GoogleSheets
+  # @api private
+  class Config
+    FIELDS = %w[client_id client_secret scope refresh_token type].freeze
+    attr_accessor(*FIELDS)
+
+    def initialize(config_path)
+      @config_path = config_path
+      if ::File.exist?(config_path)
+        JSON.parse(::File.read(config_path)).each do |key, value|
```

Comment:
When testing this locally, I had to change this to line to `JSON.parse(::File.read(config_path))['installed']` since the client configuration json (`credentials.json`) that you download [here](https://developers.google.com/sheets/api/quickstart/ruby#step_1_turn_on_the_api_name) has all of the config nested under the `installed` key.